### PR TITLE
Unit test/items py

### PR DESCRIPTION
**Pull request by mugdhadhole1**
Unit tests for items.py file

### CODE REVIEW

File: test-unit/test_items.py [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+                    self.assertEqual(result.location.gh_root, location_data["gh_root"])
+                    self.assertEqual(result.location.commit, location_data["commit"])
+                    self.assertEqual(result.location.filename, location_data["file"])
+                    self.assertEqual(result.location.line, location_data["line"])
+                elif location_type == "codebeamer":
+                    self.assertEqual(result.location.cb_root, location_data["cb_root"])
+                    self.assertEqual(result.location.tracker, location_data["tracker"])
+                    self.assertEqual(result.location.item, location_data["item"])
+                    self.assertEqual(result.location.version, location_data["version"])
+                    self.assertEqual(result.location.name, location_data["name"])
```
Comment (kedarnn) [inline test-unit/test_items.py:111]: Few observations over here

1. This class is basically used to test the LOBSTER items classes, so my suggestion would be to create separate test case for every LOBSTER item class. Same implementation as that is done in classes TestRequirement or TestItem etc.
2. The class name can be more relevant to the tests it has. Setup is generally used as a test fixture to prepare and do the pre test activities.

File: test-system/lobster-python/basic.output [hunk 2]
```diff
@@ -63,4 +63,3 @@ Written output for 3 items to basic.lobster
   "schema": "lobster-imp-trace",
   "version": 3
 }
-
```
Comment (phiwuu) [inline test-system/lobster-python/basic.output:66]: Is this change really necessary?

File: test-system/lobster-python/pytest_mark.output [hunk 3]
```diff
@@ -48,4 +48,3 @@ Written output for 2 items to pytest_mark.lobster
   "schema": "lobster-act-trace",
   "version": 3
 }
-
```
Comment (phiwuu) [inline test-system/lobster-python/pytest_mark.output:51]: Is this change really necessary?

**Review by phiwuu — CHANGES_REQUESTED** [review]

File: test-unit/test_items.py [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+    @patch("lobster.items.Tracing_Tag.key")
+    def test_determine_status_ok(self, mock_key):
+        SetUp.item.ref_up = []
+        SetUp.item.ref_down = [SetUp.tracing_tag]
+        SetUp.item.just_up = []
+        SetUp.item.just_down = []
+        SetUp.item.just_global = []
+        SetUp.item.messages = []
+        SetUp.item.has_error = False
+        SetUp.item.level = "level1"
```
Comment (kedarnn) [inline test-unit/test_items.py:171]: Just inherit the SetUp class and access these through self.

File: test-unit/test_items.py
Comment (kedarnn) [inline test-unit/test_items.py:1]: We should add blank lines to separate logical sections and avoid adding unnecessary blank lines in-between the code.

File: lobster/tools/python/python.py [hunk 5]
```diff
@@ -500,7 +500,6 @@ def main():
     if options.out:
         with open(options.out, "w", encoding="UTF-8") as fd:
             lobster_write(fd, schema, "lobster_python", items)
-            fd.write("\n")
```
[the remark this answers is not in the document]
Comment (mugdhadhole1) [reply lobster/tools/python/python.py:503]: As I have mentioned in the commit message,
Previously make test-ci command added two empty lines at the end of the content in output files now you will see only one empty line at the end of the content in those output files

File: test-system/lobster-python/pytest_mark.output (hunk 3 above)
Comment (mugdhadhole1) [reply test-system/lobster-python/pytest_mark.output:51]: Previously make test-ci command added two empty lines at the end of the content in output files 
I made this change to remove the extra lines.

In recent commit, I have removed the code which added extra empty line to the end of content in output files,
now you will see only one empty line at the end of content in those output files

File: test-system/lobster-python/basic.output (hunk 2 above)
Comment (mugdhadhole1) [reply test-system/lobster-python/basic.output:66]: Previously make test-ci command added two empty lines at the end of the content in output files
I made this change to remove the extra lines.

In recent commit, I have removed the code which added extra empty line to the end of content in output files,
now you will see only one empty line at the end of content in those output files

File: lobster/tools/python/python.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (kedarnn) [reply lobster/tools/python/python.py:503]: I guess this comment is not relevant to this review comment. I think you wanted to add it for the one which Philipp had added.

[the remark this answers is not in the document]
Comment (mugdhadhole1) [reply lobster/tools/python/python.py:503]: To elaborate my comment,
This specific code line `fd.write("\n")` added two empty lines at the end of the content in output files.
Hence, after running the make test-ci command, two empty lines were added at the end of the content in output files, now that I have removed the `fd.write("\n")` you will see only one empty line at the end of the content in those output files.

[the remark this answers is not in the document]
Comment (kedarnn) [reply lobster/tools/python/python.py:503]: Oh now I get it. I misunderstood your comment.

**Review by phiwuu — APPROVED** [review]
